src/OpenLoco/Vehicles/VehicleBody.cpp

```diff
@@ -948,28 +942,14 @@ namespace OpenLoco::Vehicles
         }
 
         var_05 += 64;
-        Map::Pos3 loc = {
-            static_cast<int16_t>(backBogie->x - frontBogie->x),
-            static_cast<int16_t>(backBogie->y - frontBogie->y),
-            static_cast<int16_t>(backBogie->z - frontBogie->z),
-        };
-
-        loc.x = loc.x * var_05 / 128;
-        loc.y = loc.y * var_05 / 128;
-        loc.z = loc.z * var_05 / 128;
-
-        loc.x += frontBogie->x;
-        loc.y += frontBogie->y;
-        loc.z += frontBogie->z;
-
-        loc.z += vehicleObject->animation[num].height;
 
         auto xyFactor = Math::Trigonometry::computeXYVector(vehicleObject->animation[num].height, sprite_pitch, sprite_yaw);
 
-        loc.x += xyFactor.x;
-        loc.y += xyFactor.y;
+        auto bogieDifference = frontBogie->position - backBogie->position;
 
-        Exhaust::create(loc, vehicleObject->animation[num].object_id | (soundCode ? 0 : 0x80));
+        auto smokeLoc = bogieDifference * var_05 / 128 + frontBogie->position + Map::Pos3{ static_cast<coord_t>(xyFactor.x), static_cast<coord_t>(xyFactor.y), vehicleObject->animation[num].height };
```

Comment:
Can we avoid these casts by using the following?
```suggestion
        auto smokeLoc = bogieDifference * var_05 / 128 + frontBogie->position + Map::Pos3(xyFactor.x, xyFactor.y, vehicleObject->animation[num].height);
```